### DISCUSSION TITLE
2.8" TFT 3-line SPI

### DISCUSSION THREAD

**Discussion by Plombir18** [Bodmer/TFT_eSPI#1771] [Compatible displays and setup files]
Hi guys,

I would like to use the TFT_eSPI lib on my ESP32 with a 2.8" TFT display with a 3 line SPI and ST7789V chip. Anyway, I'm struggling with the setup and I'm not sure if TFT_eSPI supports the 3-line SPI interface.

The pins I defined in the User_Setup.h are the following uncommented defines:
#define ST7789_DRIVER      // Full configuration option, define additional parameters below for this display
#define TFT_SDA_READ      // This option is for ESP32 ONLY, tested with ST7789 and GC9A01 display only
#define TFT_RGB_ORDER TFT_RGB  // Colour order Red-Green-Blue
#define TFT_WIDTH  240 // ST7789 240 x 240 and 240 x 320
#define TFT_HEIGHT 320 // ST7789 240 x 320

#define TFT_MOSI 27 // In some display driver board, it might be written as "SDA" and so on.
#define TFT_SCLK 26
#define TFT_CS   4  // Chip select control pin
//#define TFT_DC   27  // Data Command control pin
#define TFT_RST  25  // Reset pin (could connect to Arduino RESET pin)

The TFT_DC is still commented as I only have 3 lines. So the basic question now is, does the library support such types of tft displays?

Defined pins of the display:
![grafik](https://user-images.githubusercontent.com/103451681/162803581-82c04a45-a87e-4186-b719-9b310bacb8be.png)

Thanks in advance for any help :)

**Comment by Bodmer** [comment]
The library does not support 3 line SPI displays. The DC (Data/Command Register Select) line is needed. The 3 line SPI displays require the DC signal to be sent as an extra bit in the serial data and the library does not have a mode for that.

**Reply by Plombir18** [reply]
Understood, thanks for the clarification.

Is it possible to add the 3-line SPI support for the library? I am willing to donate this extension if technically possible.

**Reply by Bodmer** [reply]
Hi, yes it would be possible but it would require a significant effort to make the changes and test them for all the supported processors. Most controllers support a 4 wire interface but on pre-built boards this option is not always available.

The good news is that there are plenty of 4 wire SPI displays available that are supported.

**Reply by Plombir18** [reply]
OK, so do you think you will add the support for the 3-wire SPI?

**Reply by Bodmer** [reply]
Not unless the market gets flooded with 3 wire displays and 4 wire ones are hard to find.

**Reply by Plombir18** [reply]
Is it possible that you customize the lib for a fee for 3-wire SPI? How much would be the fee?
